File: Functions/rock_paper_scissor/rockPaperScissor.js
//Rock Paper Scissor game
//Rules:
//Can only chose between rock, paper, scissor (points are 1, 2, 3 respectively)
//Won = 6
//Tie = 3
//Loss = 0

//X,Y,Z = Rock, paper, scissor (For me)
//A,B,C = Rock, paper, scissor (Opponent)

//Objective
//There is a guide that the elves have given that shows the prediction of each players options
//What is the total point for me based on the strategy guide. 



const { match } = require('assert');
const data = require('../../Master_Data/rock_paper_scissor/rockPaperScissorData');
let finalProductionData = { opponent: { action: [], score: 0, roundMatchResult: "" }, me: { action: [], score: 0, roundMatchResult: "" } };



//function to convert json data into formatted object for easier accessability
const formatData = () => {
  splitCharactersArray = [];
  splitCharactersArray = data.split(",");
  console.log('splitCharactersArray', splitCharactersArray);
  //Loop thru the splitCharacterArray and if the value index is even = opponent, if odd = me
  splitCharactersArray.forEach((values, index) => {
    index % 2 == 0 ? finalProductionData.opponent.action.push(values) : finalProductionData.me.action.push(values)
  });
};


formatData();

//Function to calculate the action taken from competitors (rock, paper, and scissor values)
const actionComparison = (recordedActionData) => {
  //Object storing data about how much each action is worth 
  // A and X == rock
  // B and Y == paper
  // C and Z == scissor
  const actionPoints = { "A": 1, "B": 2, "C": 3, "X": 1, "Y": 2, "Z": 3 };

  //Loop thru opponent action and compare it to action points library
  recordedActionData.opponent.action.forEach((opponentAction, index) => {
    recordedActionData.opponent.score += actionPoints[opponentAction];
   
    //Since the number of matches are the same, we can pass the index position from opponent directly into myAction
    const myAction = recordedActionData.me.action[index];
    recordedActionData.me.score += actionPoints[myAction];

  
    //Calculate match points
    calculateMatchPoints(opponentAction, myAction);
  });
};

//Function to calculate match points
const calculateMatchPoints = (opponentAction, myAction) => {
  //Logic to detminer win, loss, and tie
  if ((opponentAction === 'A' && myAction ==='X') || (opponentAction === 'B' && myAction ==='Y') || (opponentAction === 'C' && myAction ==='Z')) {
    finalProductionData.opponent.roundMatchResult = "tie";
    finalProductionData.me.roundMatchResult = "tie";
  } else if ((opponentAction === 'A' && myAction ==='Z') || (opponentAction === 'B' && myAction ==='X') || (opponentAction === 'C' && myAction ==='Y')) {
    finalProductionData.opponent.roundMatchResult = "win";
    finalProductionData.me.roundMatchResult = "loss";
  } else if((opponentAction === 'A' && myAction ==='Y') || (opponentAction === 'B' && myAction ==='Z') || (opponentAction === 'C' && myAction ==='X')) {
    finalProductionData.opponent.roundMatchResult = "loss";
    finalProductionData.me.roundMatchResult = "win";
  }

  //Object storing data about how much win loss and tie are worth in points
  const roundMatchResult = { "win": 6, "loss": 0, "tie": 3 };

  const opponentroundMatchResult = finalProductionData.opponent.roundMatchResult;
  const myroundMatchResult = finalProductionData.me.roundMatchResult;

  //Find the value of the match result points from roundMatchResult object and add it to both opponent and my score
  finalProductionData.opponent.score += roundMatchResult[opponentroundMatchResult];
  finalProductionData.me.score += roundMatchResult[myroundMatchResult];

};



actionComparison(finalProductionData);
console.log(finalProductionData)

File: Functions/rock_paper_scissor/rockPaperScissor.rb


require_relative "../../Master_Data/rock_paper_scissor/rock_paper_scissor.rb"
include Master_data

$strat_guide = Master_data.format_data

#Rock Paper Scissor game
#Rules:
#Can only chose between rock, paper, scissor (points are 1, 2, 3 respectively)
#Won = 6
#Tie = 3
#Loss = 0

#X,Y,Z = Rock, paper, scissor (For me)
#A,B,C = Rock, paper, scissor (Opponent)

#Objective
#There is a guide that the elves have given that shows the prediction of each players options
#What is the total point for me based on the strategy guide. 

$final_production_data = {opponent:{action:[], score:0, round_match_result: ""},  me: { action: [], score: 0, roundMatchResult: "" }}

# Function to convert json data into formatted object for easier accessability
def format_data() 

  split_characters_array =  $strat_guide

  # Loop thru the split_character_Array and if the value index is even = opponent, if odd = me (currently at 4800 characters)
  split_characters_array.each_with_index do |character, index|
    index % 2 == 0 ? $final_production_data[:opponent][:action].push(character) : $final_production_data[:me][:action].push(character)
  end
end

format_data()

#Function to calculate the action taken from competitors (rock, paper, and scissor values)
def action_comparison()
  #Object storing data about how much each action is worth 
  # A and X == rock
  # B and Y == paper
  # C and Z == scissor
  action_points = {A: 1, B: 2, C: 3, X: 1, Y: 2, Z: 3}

  #Loop thru opponent action and compare it to action points library
  $final_production_data[:opponent][:action].each_with_index do |opponent_action, index|
    $final_production_data[:opponent][:score] += action_points[opponent_action.to_sym].to_i
   
    # Since the number of matches are the same, we can pass the index position from opponent directly into myAction
    my_action = $final_production_data[:me][:action][index];
    $final_production_data[:me][:score] += action_points[my_action.to_sym]

    # Calculate match points
    calculate_match_points(opponent_action, my_action)
  end
end

# Function to calculate match points
####Perhaps make two methods? for evaulating and calculating match result points
def calculate_match_points(opponent_action, my_action) 
  #Logic to determine win, loss, or tie
  #####LOOK INTO HASH LOOKUP!!!!
  if ((opponent_action == 'A' && my_action == 'X') || (opponent_action == 'B' && my_action == 'Y') || (opponent_action == 'C' && my_action == 'Z'))
    $final_production_data[:opponent][:round_match_result] = "tie"
    $final_production_data[:me][:round_match_result] = "tie"

  elsif (((opponent_action == 'A' && my_action == 'Z') || (opponent_action == 'B' && my_action == 'X') || (opponent_action == 'C' && my_action == 'Y')))
    $final_production_data[:opponent][:round_match_result] = "win"
    $final_production_data[:me][:round_match_result] = "loss"
    
  elsif (((opponent_action == 'A' && my_action == 'Y') || (opponent_action == 'B' && my_action == 'Z') || (opponent_action == 'C' && my_action == 'X')))
    $final_production_data[:opponent][:round_match_result] = "loss"
    $final_production_data[:me][:round_match_result] = "win"
  end
  #Object storing data about how much win loss and tie are worth in points
  round_match_result ={ win: 6, loss: 0, tie: 3 }
  
  opponent_round_match_result = $final_production_data[:opponent][:round_match_result]
  my_round_match_result = $final_production_data[:me][:round_match_result]

  #Find the value of the match result points from roundMatchResult object and add it to both opponent and my score
  $final_production_data[:opponent][:score] += round_match_result[opponent_round_match_result.to_sym]
  $final_production_data[:me][:score] += round_match_result[my_round_match_result.to_sym]
  
  puts $final_production_data[:me][:score]
end

action_comparison()